nomad/resource_quota_specification.go

```diff
@@ -0,0 +1,270 @@
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceQuotaSpecification() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceQuotaSpecificationCreate,
+		Update: resourceQuotaSpecificationUpdate,
+		Delete: resourceQuotaSpecificationDelete,
+		Read:   resourceQuotaSpecificationRead,
+		Exists: resourceQuotaSpecificationExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Unique name for this quota specification.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+
+			"description": {
+				Description: "Description for this quota specification.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+
+			"limits": {
+				Description: "Limits encapsulated by this quota specification.",
+				Required:    true,
+				Type:        schema.TypeSet,
+				Elem:        resourceQuotaSpecificationLimits(),
+			},
+		},
+	}
+}
+
+func resourceQuotaSpecificationLimits() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Description: "Region in which this limit has affect.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"region_limit": {
+				Required:    true,
+				Description: "The limit applied to this region.",
+				Type:        schema.TypeSet,
+				MaxItems:    1,
+				Elem:        resourceQuotaSpecificationRegionLimits(),
+			},
+		},
+	}
+}
+
+func resourceQuotaSpecificationRegionLimits() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"cpu": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"memory_mb": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceQuotaSpecificationCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	spec := api.QuotaSpec{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+	}
+	limits, err := expandQuotaLimits(d)
+	if err != nil {
+		return err
+	}
+	spec.Limits = limits
+
+	// upsert our quota spec
```

Comment:
Since this is an upsert like the update, could you just called the update here and then `SetId`? Or add some shared inner func? They appear to be the same code except for some messaging and the `SetId`.